src/graph.rs

```diff
@@ -111,56 +105,60 @@ impl<'tree> Graph<'tree> {
                 let ctx = self.1;
                 let mut seq = serializer.serialize_seq(Some(graph.graph_nodes.len()))?;
                 for (node_index, node) in graph.graph_nodes.iter().enumerate() {
-                    seq.serialize_element(&JSONNode(node_index, node, ctx))?;
+                    seq.serialize_element(&InContext(&(node_index, node), ctx))?;
```

Comment:
Not sure if this is a good idea, but you could implement a method on `Context` to create these `InContext` values, so that you could write `ctx.with(&(node_index, node))`.

Reply:
I dig it 👍🏻